File: app/src/main/java/lt/markmerkk/DBTestExecutor.java
package lt.markmerkk;

import java.io.IOException;
import java.net.URI;
import java.net.URISyntaxException;
import java.nio.file.FileSystems;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import lt.markmerkk.entities.database.DBBaseExecutor;

/**
 * Created by mariusmerkevicius on 11/21/15.
 * Test database for various mock sqlite transactions
 */
public class DBTestExecutor extends DBBaseExecutor {

  public static final String FILE = "test_database.db";

  public DBTestExecutor() {
    try {
      Path path = FileSystems.getDefault().getPath(FILE);
      boolean result = Files.deleteIfExists(path);
    } catch (IOException e) {
      e.printStackTrace();
    }
    migrate();
  }


  @Override
  protected String database() {
    return "test_database.db";
  }

  @Override
  protected URI migrationScriptPath() {
    return Paths.get("src/main/resources/" + "changelog_1.xml").toUri();
  }

  @Override
  protected URI migrationExportPath() {
    try {
      return getClass().getResource("/").toURI();
    } catch (URISyntaxException e) {
      return null;
    }
  }


}

File: app/src/main/java/lt/markmerkk/DBTestExecutor.kt
package lt.markmerkk

import lt.markmerkk.entities.LocalIssue
import lt.markmerkk.entities.SimpleLog
import java.io.IOException
import java.net.URI
import java.net.URISyntaxException
import java.nio.file.FileSystems
import java.nio.file.Files
import java.nio.file.Paths
import lt.markmerkk.entities.database.DBBaseExecutor
import lt.markmerkk.entities.jobs.CreateJobIfNeeded
import org.slf4j.LoggerFactory

/**
 * Created by mariusmerkevicius on 11/21/15.
 * Test database for various mock sqlite transactions
 */
class DBTestExecutor : DBBaseExecutor() {

    init {
        logger.debug("Applying database migrations")
        try {
            val path = FileSystems.getDefault().getPath(FILE)
            val result = Files.deleteIfExists(path)
            executeOrThrow(CreateJobIfNeeded(SimpleLog::class.java))
            executeOrThrow(CreateJobIfNeeded(LocalIssue::class.java))
        } catch (e: IOException) {
            e.printStackTrace()
        }
    }


    override fun database(): String {
        return "test_database.db"
    }

    override fun migrationScriptPath(): URI? {
        return null
    }

    override fun migrationExportPath(): URI? {
        return null
    }

    companion object {
        private val logger = LoggerFactory.getLogger(DBTestExecutor::class.java)!!
        private val FILE = "test_database.db"
    }

}
